x/evm/keeper/vote_handler.go

```diff
@@ -144,6 +144,7 @@ func (v voteHandler) HandleResult(ctx sdk.Context, result codec.ProtoMarshaler)
 	if err != nil {
 		// set events to failed, we will deal with later
 		for _, e := range voteEvents.Events {
+			// TODO: handle error
```

Comment:
Why not return early in this case since if this occurs, the state is corrupt?

Reply:
Returning error here is still gonna corrupt the code because of how the vote/msg_server is currently using it. The goal of this PR isn't to fix this/change behaviour but just more like a refactor.